constants.go

```diff
@@ -0,0 +1,72 @@
+package gojsonsm
+
+import (
+	"fmt"
+)
+
+// Function related constants
+const (
+	MathFuncAbs   string = "mathAbs"
+	MathFuncAcos  string = "mathFuncAcos"
+	MathFuncAsin  string = "mathFuncAsin"
+	MathFuncAtan  string = "mathFuncAtan"
+	MathFuncCeil  string = "mathCeil"
+	MathFuncCos   string = "mathCos"
+	MathFuncExp   string = "mathFuncExp"
+	MathFuncFloor string = "mathFuncFloor"
+	MathFuncLog   string = "mathFuncLog"
+	MathFuncLn    string = "mathFuncLn"
+	MathFuncRound string = "mathRound"
+	MathFuncSin   string = "mathSin"
+	MathFuncSqrt  string = "mathSqrt"
+	MathFuncTan   string = "mathTan"
+)
+
+// Error constants
+var emptyExpression Expression
+var ErrorNotFound error = fmt.Errorf("Error: Specified resource was not found")
+var ErrorNoMoreTokens error = fmt.Errorf("Error: No more token found")
+var ErrorNeedToStartOneNewCtx error = fmt.Errorf("Error: Need to spawn one subcontext")
+var ErrorNeedToStartNewCtx error = fmt.Errorf("Error: Need to spawn subcontext")
+var ErrorParenMismatch error = fmt.Errorf("Error: Parenthesis mismatch")
+var ErrorParenWSpace error = fmt.Errorf("Error: parenthesis must have white space before or after it")
+var NonErrorOneLayerDone error = fmt.Errorf("One layer has finished")
+var ErrorLeadingZeroes error = fmt.Errorf("Nested mode index must not have leading zeros")
+var ErrorAllInts error = fmt.Errorf("Array index must be a valid integer")
+var ErrorEmptyNest error = fmt.Errorf("Array index cannot be empty")
+var ErrorMissingBacktickBracket error = fmt.Errorf("Invalid field - could not find matching ending backtick or bracket")
+var ErrorEmptyLiteral error = fmt.Errorf("Literals cannot be empty")
+var ErrorEmptyToken error = fmt.Errorf("Token cannot be empty")
+var ErrorInvalidFuncArgs error = fmt.Errorf("Unable to parse arguments to specified built in function")
+var ErrorPcreNotSupported error = fmt.Errorf("Error: Current instance of gojsonsm does not have native PCRE support compiled")
```

Comment:
This doesn't seem related to adding math functions?

Reply:
A new commit within this PR is used to describe the creation of a constants.go file, so this comment is addressed.